src/test/java/org/assertj/core/extention/SoftlyExtensionExecutionTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2020 the original author or authors.
+ */
+package org.assertj.core.extention;
+
+import org.assertj.core.annotations.Softly;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.api.ThrowableAssert.ThrowingCallable;
+import org.assertj.core.extention.SoftlyExtension;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.assertj.core.util.AssertionsUtil.assertThatAssertionErrorIsThrownBy;
+
+@DisplayName("SoftlyExtension runs")
+@ExtendWith({SoftlyExtension.class, MockitoExtension.class})
+class SoftlyExtensionExecutionTest {
+
+  @Softly(assertAfter = true)
+  @InjectMocks
+  private SoftAssertions softlyAssertAfterTrue;
+
+  @Softly(assertAfter = false)
```

Comment:
I don't see the interest of this use case

Reply:
Someone might just need the `@Sofly` field injections but they want control of when `#assertAll()` is invoked .

Reply:
I don't see any use case and it's error prone since the user might forget to call `assertAll` (or remove it later by error) : **that would give false positive where a test is passing instead of failing**.

Reply:
I think the annotation has some value this is because it shows intent. I think for field injection intent is very important. For example, when using field dependency injection, we must provide the `@Autowired` annotation. Also, I was inspired by Mockito's `@Captor` annotation.

Reply:
I don't see the point of `assertAfter = false`, I mean if people want to control soft assertions verification phase, they can just create a regular SoftAssertions with `new` and call `assertAll()` when needed.